src/Compilers/CSharp/Portable/Binder/Binder_Lambda.cs

```diff
@@ -94,6 +94,14 @@ internal partial class Binder
 
                 foreach (var p in parameterSyntaxList.Value)
                 {
+                    if (p.AttributeLists.Count > 0 && syntax.Kind() == SyntaxKind.AnonymousMethodExpression)
```

Comment:
Looks like the parser does not even support them on lambdas:

```c#
        private bool IsPossibleLambdaParameter()
        {
            switch (this.CurrentToken.Kind)
            {
                case SyntaxKind.ParamsKeyword:
                // params is not actually legal in a lambda, but we allow it for error
                // recovery purposes and then give an error during semantic analysis.
                case SyntaxKind.RefKeyword:
                case SyntaxKind.OutKeyword:
                case SyntaxKind.OpenParenToken:   // tuple
                    return true;

                case SyntaxKind.IdentifierToken:
                    return this.IsTrueIdentifier();

                default:
                    return IsPredefinedType(this.CurrentToken.Kind);
            }
        }
```

So we don't end up in the situation where we would get a lambda with attributes in the syntax tree currently.